include/ua_plugin_securitypolicy.h

```diff
@@ -0,0 +1,424 @@
+                                                void *endpointContext);
+
+    const UA_ByteString *(*const getServerCertificate)(const UA_SecurityPolicy *securityPolicy,
+                                                       const void *endpointContext);
+
+    UA_StatusCode (*const setCertificateTrustList)(const UA_SecurityPolicy *securityPolicy,
+                                                   const UA_ByteString *trustList,
+                                                   void *endpointContext);
+
+    UA_StatusCode (*const setCertificateRevocationList)(const UA_SecurityPolicy *securityPolicy,
+                                                        const UA_ByteString *revocationList,
+                                                        void *endpointContext);
+
+    /**
+    * Gets the signature size that depends on the local private key.
+    * This will return 0 as long as no remote certificate was set.
+    */
+    size_t (*const getLocalAsymSignatureSize)(const UA_SecurityPolicy *securityPolicy,
+                                              const void *endpointContext);
+
+    /**
+     * \brief Compares the supplied certificate with the certificate in the endpoit context.
+     *
+     * \param securityPolicy contains the function pointers associated with the policy.
+     * \param endpointContext the endpoint context data that contains the certificate to compare to.
+     * \param certificateThumbprint the certificate thumbprint to compare to the one stored in the context.
+     * \return if the thumbprints match UA_STATUSCODE_GOOD is returned. If they don't match
+     *         or an errror occured an error code is returned.
+     */
+    UA_StatusCode (*const compareCertificateThumbprint)(const UA_SecurityPolicy *securityPolicy,
+                                                        const void *endpointContext,
+                                                        const UA_ByteString *certificateThumbprint);
+};
+
+struct UA_Channel_SecurityContext {
+    /**
+     * \brief This method initializes a new context data object.
+     * The caller needs to call deleteMembers on the recieved object to free allocated memory.
+     *
+     * Memory is only allocated if the function succeeds so there is no need to manually free
+     * the memory pointed to by *pp_contextData or to call deleteMembers.
+     *
+     * \param securityPolicy contains the function pointers associated with the policy.
+     * \param remoteCertificate the remote certificate contains the remote asymmetric key.
+     *                          The certificate will be verified and then stored in the context
+     *                          so that its details may be accessed.
+     * \param contextData the initialized contextData that is passed to functions that work
+     *                    on a context.
+     */
+    UA_StatusCode (*const init)(const UA_SecurityPolicy *securityPolicy,

```

Comment:
`summon`
(a linked list from the void, https://aphyr.com/posts/342-typing-the-technical-interview).

Just kidding.

`newContext` and `deleteContext` goes with the existing verbiage.